### ISSUE TITLE
If I use the pretrained model but the input-dim isn't match ,how can i adjust input-dim of my wavs?

### ISSUE THREAD

**Issue by baowenqian2001** [kaldi-asr/kaldi#4843]
ERROR (nnet3-compute[5.5.1061~2-e4eb]:EnsureFrameIsComputed():nnet-am-decodable-simple.cc:101) Neural net expects 'input' features with dimension 43 but you provided 40

**Comment by desh2608** [comment]
You should use the same feature config file (`conf/mfcc.conf`) to extract features for your wav, as the one which was used to train the model. In this case, it seems the model was trained using hires+pitch features, but you are just providing it hires features. You should use `make_mfcc_pitch.sh` to extract features instead of `make_mfcc.sh`. You would also need to add a `pitch.conf` in your `conf` directory (search for this file in kaldi/egs for an example).

**Comment by baowenqian2001** [comment]
Appreciate it so much. It's useful!

**Comment by baowenqian2001** [comment]
Appreciate it so much！发自我的荣耀手机-------- 原始邮件 --------发件人： Desh Raj ***@***.***>日期： 2023年4月22日周六 晚上8:38收件人： kaldi-asr/kaldi ***@***.***>抄送： baowenqian2001 ***@***.***>, Author ***@***.***>主    题： Re: [kaldi-asr/kaldi] If I use the pretrained model but the input-dim isn't match ,how can i adjust input-dim of my wavs? (Issue #4843)
You should use the same feature config file (conf/mfcc.conf) to extract features for your wav, as the one which was used to train the model. In this case, it seems the model was trained using hires+pitch features, but you are just providing it hires features. You should use make_mfcc_pitch.sh to extract features instead of make_mfcc.sh. You would also need to add a pitch.conf in your conf directory (search for this file in kaldi/egs for an example).
